### DISCUSSION TITLE
Add interactive elements within a select option

### DISCUSSION THREAD

**Discussion by AinaVendrell** [adobe/react-spectrum#6626] [Q&A] [answered]
Hi,

I've created a select component using the `useSelect` hook from React Aria. I would like to reuse this component in various parts of my application where I need to add interactive elements like links or buttons within the options. However, I'm encountering an issue: when the user clicks on the button or link, the option gets selected, and I don't receive any events in the onPress method of the button and the links won't open.

How can I implement this functionality?

Here is the [CodeSandbox](https://codesandbox.io/p/sandbox/react-aria-select-item-with-button-and-link-c9pjv6?file=%2Fsrc%2FApp.tsx%3A13%2C34) demonstrating my current implementation.

![Captura de pantalla 2024-06-26 a les 16 27 52](https://github.com/adobe/react-spectrum/assets/59646082/6fc1b0dd-4620-46f0-9f6d-9744102deb95)

**Comment by reidbarber** [comment, marked as answer]
`useSelect` is meant to be combined with `useListBox`, and the [ARIA listbox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/) does not support having interactive elements within items. [useGridList](https://react-spectrum.adobe.com/react-aria/useGridList.html) does support interactive elements within items, but it isn't meant to be combined with `useSelect`, and won't be exposed to assistive technology in the same way.
